### PR TITLE
Switch to golangci

### PR DESCRIPTION
**Pull request by lavoiesl**
Faster and better supported than gometalinter.

- Mark `TtlForExpiration` as deprecated, it does not need to be exported
- Return exported types in constructors
- Fix comments for deprecated variables

### CODE REVIEW

File: memcached.go [hunk 1]
```diff
@@ -9,9 +9,7 @@ import (
 	"github.com/bradfitz/gomemcache/memcache"
 )
 
-var _ Client = &memcacheClient{}
```
Comment (yousifh) [inline memcached.go:12]: Why was this removed?

Comment (lavoiesl) [reply memcached.go:12]: Since the constructor now returns a `Client`, this is now superfluous.

Note that the usage doesn't change, so this shouldn't affect clients.

**Review by bmassemin — APPROVED** [review]
